test-app/server/server.js

```diff
@@ -0,0 +1,36 @@
+const dotenv = require('dotenv');
+const express = require('express');
+const router = express.Router();
+var cors = require('cors');
+const app = express();
+app.use(cors());
+
+dotenv.config();
+
+const uuid = require('uuid');
+const crypto = require("crypto");
+
+const privateKey = process.env.PRIVATE_KEY;
+
+router.get("/auth", function (req, res) {
+  var token = req.query.token || uuid.v4();
+  var expire = req.query.expire || parseInt(Date.now() / 1000) + 2400;
+  var privateAPIKey = `${privateKey}`;
+  var signature = crypto.createHmac('sha1', privateAPIKey).update(token + expire).digest('hex');
+  res.status(200);
+  res.send({
+    token: token,
+    expire: expire,
+    signature: signature
+  });
+});
+router.get("/", (req, res) => {
+  res.status(200);
+  res.send();
+});
+
+app.use("/", router);
+
+app.listen(3001, function () {
+  console.log("Live at Port 3001");
+});
```

Comment:
Instead of having this as a external dependecy, can we have this as an api in next.js itself?

https://nextjs.org/docs/pages/building-your-application/routing/api-routes
https://nextjs.org/docs/app/building-your-application/routing/route-handlers

Reply:
done